src/error.rs

```diff
@@ -108,6 +120,26 @@ impl Message {
             column: 0,
             index: 0,
             message,
+            notes: String::new(),
+        }
+    }
+
+    #[cfg(feature = "use-compiled-tools")]
+    unsafe fn message_and_notes_from_cstr(msg: *const std::os::raw::c_char) -> (String, String) {
+        let full_message = std::ffi::CStr::from_ptr(msg).to_string_lossy();
+
+        if let Some(ind) = full_message.find('\n') {
+            (
+                full_message[..ind].to_owned(),
+                // The compiled code always adds a trailing newline, even if it
+                // is the last note, presumably since it's copied directly from
+                // the source, but we don't want it, note we don't use trim since
+                // (other than the first \n) there can be significant whitespace
+                // at the beginning
+                full_message[ind + 1..].trim_end().to_owned(),
```

Comment:
```suggestion
                full_message[ind + 1..].trim_end_matches('\n').to_owned(),
```
(assuming you want specifically to deal with only `\n`)